rclpy/rclpy/__init__.py

```diff
@@ -109,6 +110,7 @@ def spin_once(node, *, timeout_sec=None):
         if request:
             response = srv.callback(request, srv.srv_type.Response())
             srv.send_response(response, header)
+    gc.collect()
```

Comment:
Maybe the tests can either trigger the GC before finishing or monkey patch rclpy to perform this on shutdown?